src/test/java/edu/wpi/grip/core/ConnectionTest.java

```diff
@@ -47,4 +51,26 @@ public void testInputSocketNotNull() {
         Socket<Double> foo = new Socket<>(eventBus, fooHint);
         Connection<Double> connection = new Connection<>(eventBus, foo, null);
     }
+
+    @Test
+    public void testForwardsConnection() {
+        Socket<Double> output = new Socket<>(eventBus, fooHint);
+        output.setDirection(Socket.Direction.OUTPUT);
+
+        Socket<Double> input= new Socket<>(eventBus, fooHint);
+        input.setDirection(Socket.Direction.INPUT);
+
+        new Connection<Double>(eventBus, output, input);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBackwardsConnection() {
+        Socket<Double> output = new Socket<>(eventBus, fooHint);
+        output.setDirection(Socket.Direction.OUTPUT);
+
+        Socket<Double> input= new Socket<>(eventBus, fooHint);
+        input.setDirection(Socket.Direction.INPUT);
```

Comment:
To explicitly set the directions of the two sockets so we can make a connection between them.  In the application, this is done in `Step` immediately after they're created, but for the test I'm manually setting them.